### ISSUE TITLE
Global variable definition in fribidi.h leads to problems with inclusion from multiple source files

### ISSUE THREAD

**Issue by sbstnk** [fribidi/fribidi#144]
`fribidi.h` defines a global variable `fribidi_version_info` which causes linker problems when it is included from more than one source file. The same also applies to `fribidi_unicode_version` which gets defined in `fribidi-unicode.h` which gets included from `fribidi.h`.

This happens for example when compiling gtk which include `fribidi.h` in both `gdk/gdk.c` and `gdk/wayland/gdkkeys-wayland.c`. Both gtk3 (seen on 52d944ec44) and gtk4 (seen on 1b7fb10361) are affected. Here is the error message from compiling gtk3 using jhbuild on Fedora 32:

```
cc  -o gdk/libgdk-3.so.0.2404.16 'gdk/ffe01f7@@gdk-3@sha/meson-generated_.._gdkenumtypes.c.o' 'gdk/ffe01f7@@gdk-3@sha/meson-generated_.._gdkmarshalers.c.o' 'gdk/ffe01f7@@gdk-3@sha/meson-generated_.._gdkresources.c.o' 'gdk/ffe01f7@@gdk-3@sha/gdk-private.c.o' 'gdk/ffe01f7@@gdk-3@sha/gdk.c.o' 'gdk/ffe01f7@@gdk-3@sha/gdkapplaunchcontext.c.o' 'gdk/ffe01f7@@gdk-3@sha/gdkcairo.c.o' 'gdk/ffe01f7@@gdk-3@sha/gdkcursor.c.o' 'gdk/ffe01f7@@gdk-3@sha/gdkdeprecated.c.o' 'gdk/ffe01f7@@gdk-3@sha/gdkdevice.c.o' 'gdk/ffe01f7@@gdk-3@sha/gdkdevicemanager.c.o' 'gdk/ffe01f7@@gdk-3@sha/gdkdevicepad.c.o' 'gdk/ffe01f7@@gdk-3@sha/gdkdisplay.c.o' 'gdk/ffe01f7@@gdk-3@sha/gdkdisplaymanager.c.o' 'gdk/ffe01f7@@gdk-3@sha/gdkdnd.c.o' 'gdk/ffe01f7@@gdk-3@sha/gdkevents.c.o' 'gdk/ffe01f7@@gdk-3@sha/gdkframetimings.c.o' 'gdk/ffe01f7@@gdk-3@sha/gdkgl.c.o' 'gdk/ffe01f7@@gdk-3@sha/gdkglcontext.c.o' 'gdk/ffe01f7@@gdk-3@sha/gdkglobals.c.o' 'gdk/ffe01f7@@gdk-3@sha/gdkkeys.c.o' 'gdk/ffe01f7@@gdk-3@sha/gdkkeyuni.c.o' 'gdk/ffe01f7@@gdk-3@sha/gdkoffscreenwindow.c.o' 'gdk/ffe01f7@@gdk-3@sha/gdkframeclock.c.o' 'gdk/ffe01f7@@gdk-3@sha/gdkframeclockidle.c.o' 'gdk/ffe01f7@@gdk-3@sha/gdkpango.c.o' 'gdk/ffe01f7@@gdk-3@sha/gdkpixbuf-drawable.c.o' 'gdk/ffe01f7@@gdk-3@sha/gdkprofiler.c.o' 'gdk/ffe01f7@@gdk-3@sha/gdkproperty.c.o' 'gdk/ffe01f7@@gdk-3@sha/gdkrectangle.c.o' 'gdk/ffe01f7@@gdk-3@sha/gdkrgba.c.o' 'gdk/ffe01f7@@gdk-3@sha/gdkscreen.c.o' 'gdk/ffe01f7@@gdk-3@sha/gdkselection.c.o' 'gdk/ffe01f7@@gdk-3@sha/gdkvisual.c.o' 'gdk/ffe01f7@@gdk-3@sha/gdkwindow.c.o' 'gdk/ffe01f7@@gdk-3@sha/gdkwindowimpl.c.o' 'gdk/ffe01f7@@gdk-3@sha/gdkseat.c.o' 'gdk/ffe01f7@@gdk-3@sha/gdkseatdefault.c.o' 'gdk/ffe01f7@@gdk-3@sha/gdkdevicetool.c.o' 'gdk/ffe01f7@@gdk-3@sha/gdkdrawingcontext.c.o' 'gdk/ffe01f7@@gdk-3@sha/gdkmonitor.c.o' 'gdk/ffe01f7@@gdk-3@sha/deprecated_gdkcolor.c.o' -L/opt/gnome/lib -Wl,--as-needed -Wl,--no-undefined -shared -fPIC -Wl,--start-group -Wl,-soname,libgdk-3.so.0 -Wl,--whole-archive gdk/broadway/libgdk-broadway.a gdk/wayland/libgdk-wayland.a gdk/x11/libgdk-x11.a -Wl,--no-whole-archive -fno-omit-frame-pointer -Wl,-Bsymbolic -Wl,-z,relro -Wl,-z,now -lm /opt/gnome/lib/libgdk_pixbuf-2.0.so /opt/gnome/lib/libgobject-2.0.so /opt/gnome/lib/libglib-2.0.so /opt/gnome/lib/libcairo.so /opt/gnome/lib/libpango-1.0.so /opt/gnome/lib/libharfbuzz.so /opt/gnome/lib/libfribidi.so /opt/gnome/lib/libcairo-gobject.so /opt/gnome/lib/libepoxy.so /usr/lib64/libfontconfig.so /usr/lib64/libfreetype.so /opt/gnome/lib/libgio-2.0.so /opt/gnome/lib/libpangocairo-1.0.so -lrt /usr/lib64/libxkbcommon.so /opt/gnome/lib/libwayland-client.so /opt/gnome/lib/libwayland-cursor.so /opt/gnome/lib/libwayland-egl.so /usr/lib64/libXrender.so /usr/lib64/libX11.so /usr/lib64/libXi.so /usr/lib64/libXext.so /usr/lib64/libXcursor.so /usr/lib64/libXdamage.so /usr/lib64/libXfixes.so /usr/lib64/libXcomposite.so /usr/lib64/libXrandr.so /usr/lib64/libXinerama.so -Wl,--end-group
/usr/bin/ld: gdk/wayland/libgdk-wayland.a(gdkkeys-wayland.c.o):/opt/gnome/include/fribidi/fribidi.h:122: multiple definition of `fribidi_version_info'; gdk/ffe01f7@@gdk-3@sha/gdk.c.o:/opt/gnome/include/fribidi/fribidi.h:122: first defined here
/usr/bin/ld: gdk/wayland/libgdk-wayland.a(gdkkeys-wayland.c.o):/opt/gnome/include/fribidi/fribidi-unicode.h:48: multiple definition of `fribidi_unicode_version'; gdk/ffe01f7@@gdk-3@sha/gdk.c.o:/opt/gnome/include/fribidi/fribidi-unicode.h:48: first defined here
```

In gtk this can be solved by removing the include from `gdkkeys-wayland.c` where it is not really used. The gtk developers however also suggested that this maybe should also be addressed from the fribidi side to allow the inclusion of `fribidi.h` from multiple source files.

**Comment by vtorri** [comment]
try with PR #145 , please, this symbol should be correctly decorated with extern, now

**Comment by sbstnk** [comment]
When pulling master I got asked for a merge, which is weird, because I had no local changes, that I'm aware of. When looking at the git logs it seems that master on my clone of the repo had PR #138 merged, which seems to have caused the issue due to the missing `extern`, but after a "git reset --hard origin/master" the issue was gone. Was this PR removed from the git history with a forced push or something like that? Otherwise I'm not sure how I ended up with that PR merged in my clone of the repo.

Anyway, current master is working fine and with #145 applied it keeps working fine as well. Thanks!
